File: src/python/test/TestStrings.py
import unittest
from Utils import Utils 
from Refi import Refi



class TestStrings(unittest.TestCase):

    def setUp(self):
        self.refi = Refi()
        self.utils = Utils()


    def test_String(self):

        expression = self.refi.toRegEx("`ab`")
        self.utils.assertRegexIs("ab", expression)
        
        expression = self.refi.toRegEx("`!`")
        self.utils.assertRegexIs("!", expression)
        
        expression = self.refi.toRegEx("`!` `the` ` cat`")
        self.utils.assertRegexIs("!the cat", expression)
        
        expression = self.refi.toRegEx("(`!` `the` ` cat`)")
        self.utils.assertRegexIs("(!the cat)", expression)
    
    def test_UnenclosedString(self):

        Exception exception = assertThrows(RuntimeException.class, (self):
 ->
            (new Refi(self):
).toRegEx("ab"))  
        assertTrue(exception.getMessage(self):
.contains("token recognition error at: 'ab'"))

    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()    
    

File: src/test/java/koorkah/TestStrings.java
package koorkah;

import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class TestStrings extends BaseT
{

    @Test
    void testString()
    {
        String expression = koorkah.toRegEx("`ab`");
        assertRegexIs("ab", expression);
        
        expression = koorkah.toRegEx("`!`");
        assertRegexIs("!", expression);
        
        expression = koorkah.toRegEx("`!` `the` ` cat`");
        assertRegexIs("!the cat", expression);
        
        expression = koorkah.toRegEx("(`!` `the` ` cat`)");
        assertRegexIs("(!the cat)", expression);
    }
    
    @Test
    void testUnenclosedString()
    {
        Exception exception = assertThrows(RuntimeException.class, () ->
            (new Koorkah()).toRegEx("ab"));  
        assertTrue(exception.getMessage().contains("token recognition error at: 'ab'"));
    }

}
